Documentation/specs/runtime-configuration-file.md

```diff
@@ -123,6 +123,12 @@ This section is copied verbatim from an identical section in the input `project.
 
 These settings are read by `corehost` to determine how to initialize the runtime. All versions of `corehost` **must ignore** settings in this section that they do not understand (thus allowing new settings to be added in later versions).
 
+### `version` Property (`.deps.json`)
+
+This is a simple integer version indicating the version of the deps file format in use. It begins at `2` because the deps file format closely matches the format for `project.lock.json` which is at version `2`. In general, hosts should fail to load applications that use deps files with "unsupported" versions. It's up to the host to determine it's support policy. There is no guarantee of compatibility between versions of the deps file, each version could be completely different and require quirking to load properly, so no assumption should be made about the version field. The goal of this field is to allow us to make major changes without causing undefined behavior in existing hosts (instead providing a clearer "This application is not compatible with this host" error).
```

Comment:
The deps.json is processed by dotnethostimpl.dll, which is part of the SharedFramework redist. Given that, is it even possible for an application to target a version of the SharedFramework (which is already public) and yet carry a deps.json that dotnethostimpl.dll (part of the public SharedFramework that the app claims to target) does not understand?